psst-core/src/player/queue.rs

```diff
@@ -46,6 +50,23 @@ impl Queue {
         self.compute_positions();
     }
 
+    pub fn add(&mut self, item: PlaybackItem) {
+        self.added_items.push(item);
+    }
+
+    fn handle_added_queue(&mut self){
+        if self.added_items.len() > self.added_items_position {
+            self.items.insert(self.positions.len(), self.added_items[self.added_items_position]);
+            self.positions.insert(self.position + 1, self.positions.len());
+            self.added_items_position += 1;
+            
+            if self.added_items_position > 2 {
+                self.added_items.remove(0);
+                self.added_items_position -= 1;
+        }
+        }
+    }
```

Comment:
This logic is a little bit messy, could you describe what needs to happen here? Maybe there is a cleaner way of implementing this?

Reply:
My explanation for my code: The initial if statement checks if the position of the user-added queue has caught up with its length. Once that happens, the item is added to the main queue, and its position is updated in the main queue's positions.

Then, the position of the added item is incremented.

The next part ensures the vector doesn't get too long, but this isn't typically handled within the program, so I could probably remove that.

Reply:
Can we assume that adding to the queue will always increase its length? Is that first check needed?

Reply:
The first check is necessary to verify whether the user has gone through their added songs. The variable added_items_position tracks which song needs to be added next. Without this check, it would continuously attempt to add a song to the main queue, even if there isn't one available.

Sorry if the first explanation didn't make much sense, I hope this helps.